### PR TITLE
chore(deps): update dependencies

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: tests/unit/test_pebble.py
Comment (lengau) [inline tests/unit/test_pebble.py:1]: A pydantic update changed this output, so I updated the tests

File: tools/freeze-requirements.sh
Comment (lengau) [inline tools/freeze-requirements.sh:1]: Reordered this so we don't get the documentation requirements in our regular requirements.txt file.

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,131 +1,135 @@
...
 distro==1.8.0
-docutils==0.17.1
-exceptiongroup==1.0.4
-filelock==3.8.2
+docutils==0.18.1
+exceptiongroup==1.1.3
+filelock==3.12.4
 flake8==4.0.1
-furo==2022.12.7
-gnupg==2.3.1
```
Comment (tigarmo) [inline requirements-dev.txt:33]: how come this got dropped?

Comment (lengau) [reply requirements-dev.txt:33]: It was an indirect dependency via craft-archives, but we removed that dependency from craft-archives back in May.

File: tests/unit/test_pebble.py
Comment (cjdcordeiro) [reply tests/unit/test_pebble.py:1]: interesting :thinking: thanks!

Comment (lengau) [reply tests/unit/test_pebble.py:1]: It's probably not ideal that pydantic patch updates (somewhere between `1.10.0` and `1.10.13` are causing our tests to change, but that's out of scope for this PR and maybe something to ask about upstream.

**Review by tigarmo — APPROVED** [review]

**Pull request merged**